### PR TITLE
fixes to params as arrays, custom model name and from list connector …

### PR DESCRIPTION
**Pull request by chanokin**
…as non-sparse

### CODE REVIEW

File: pynn_genn/projections.py [hunk 1]
```diff
@@ -126,7 +128,14 @@ def __init__(self, presynaptic_population, postsynaptic_population,
         ContextMixin.__init__(self, {})
 
         # **TODO** leave type up to Connector types
-        self.use_sparse = (False if isinstance(connector, AllToAllConnector)
+        conn_ratio = 0.0
+        if isinstance(connector, FromListConnector) or \
```
Comment (neworderofjamie) [inline pynn_genn/projections.py]: As discussed, I don't think this quite works that simply

File: pynn_genn/__init__.py [hunk 2]
```diff
@@ -48,24 +48,25 @@ def setup(timestep=DEFAULT_TIMESTEP, min_delay=DEFAULT_MIN_DELAY,
...
-        model_name = sanitize_label(model_name)
-        model_path = os.path.dirname(calframe[1][1])
-
-        # Set model name and path (adding ./ if path is relative)
-        simulator.state.model.model_name = model_name
-        simulator.state.model_path = (model_path + os.sep
-                                      if os.path.isabs(model_path)
-                                      else "./" + model_path + os.sep)
+
+    model_name = sanitize_label(model_name)
```
Comment (neworderofjamie) [inline pynn_genn/__init__.py]: This was clearly totally broken :disappointed:

File: pynn_genn/projections.py (hunk 1 above)
Comment (chanokin) [reply pynn_genn/projections.py]: I now changed it to len(connector) == (n_pre * n_post)

File: pynn_genn/projections.py [hunk 3]
```diff
@@ -370,7 +376,20 @@ def _create_native_projection(self):
...
+            if self.use_sparse:
+                conn_params = {n: p[conn_mask] for n, p in iteritems(params)}
+            else:
+                ## GeNN stores synapses in this row-major order for dense matrices
+                ## PyNN in some cases (FromListConnector) uses column-major
+                ## thus we need to re-sort to row-major order
+                to_row_major = np.lexsort((conn_post_inds, conn_pre_inds))
+                conn_mask[:] = conn_mask[to_row_major]
+                conn_pre_inds[:] = conn_pre_inds[to_row_major]
+                conn_post_inds[:] = conn_post_inds[to_row_major]
```
Comment (neworderofjamie) [inline pynn_genn/projections.py]: If this is dense, are ``conn_pre_inds`` and ``conn_post_inds`` ever actually required again?

Comment (chanokin) [reply pynn_genn/projections.py]: True, I removed those!

**Pull request merged**